src/libcollections/vec.rs

```diff
@@ -1039,18 +1039,24 @@ impl<T> Vec<T> {
     #[inline]
     #[stable(feature = "append", since = "1.4.0")]
     pub fn append(&mut self, other: &mut Self) {
-        self.reserve(other.len());
-        let len = self.len();
-        unsafe {
-            ptr::copy_nonoverlapping(other.as_ptr(), self.get_unchecked_mut(len), other.len());
-        }
-
-        self.len += other.len();
         unsafe {
+            let count = other.len();
+            self.append_elements(other.as_ptr(), count);
             other.set_len(0);
         }
     }
 
+    /// Appends a elements to `Self`, without moving.
```

Comment:
This method does move elements read from `other` into `self`'s allocation. It should also not talk about dropping, because no `T` elements need to be dropped or are dropped; they are just moved from being stored in one buffer to the other.

Can the method `append_elements` use an argument of type `*const [T]` instead? Coercions from `&[T]` should make that simple to call in both the calling sites.